pkg/scheduler/plugins/loadaware/load_aware.go

```diff
@@ -125,40 +125,42 @@ func (p *Plugin) Filter(ctx context.Context, state *framework.CycleState, pod *c
 		}
 	}
 
-	usageThresholds, prodUsageThresholds := p.args.UsageThresholds, p.args.ProdUsageThresholds
-	customUsageThresholds, err := extension.GetCustomUsageThresholds(node)
-	if err != nil {
-		klog.V(5).ErrorS(err, "failed to GetCustomUsageThresholds from", "node", node.Name)
-	} else {
-		if len(customUsageThresholds.UsageThresholds) > 0 {
-			usageThresholds = customUsageThresholds.UsageThresholds
-		}
-		if len(customUsageThresholds.ProdUsageThresholds) > 0 {
-			prodUsageThresholds = customUsageThresholds.ProdUsageThresholds
-		}
-	}
-
-	if len(usageThresholds) > 0 {
-		status := p.filterNodeUsage(node, nodeMetric, usageThresholds)
+	filterProfile := generateUsageThresholdsFilterProfile(node, p.args)
+	if len(filterProfile.ProdUsageThresholds) > 0 && extension.GetPriorityClass(pod) == extension.PriorityProd {
+		status := p.filterProdUsage(node, nodeMetric, filterProfile.ProdUsageThresholds)
 		if !status.IsSuccess() {
 			return status
 		}
-	}
-
-	if len(prodUsageThresholds) > 0 {
-		status := p.filterProdUsage(node, nodeMetric, prodUsageThresholds)
-		if !status.IsSuccess() {
-			return status
+	} else {
```

Comment:
Here the `else` condition indicates `len(filterProfile.ProdUsageThresholds) == 0 || extension.GetPriorityClass(pod) != extension.PriorityProd`. Is it expected?

Reply:
yes. It's expected.